lib/pacemaker/pcmk_sched_clone.c

```diff
@@ -545,6 +543,7 @@ pcmk__clone_create_probe(pe_resource_t *rsc, pe_node_t *node)
         }
     }
 
+    rsc->children = g_list_sort(rsc->children, pcmk__cmp_instance_number);
```

Comment:
Is sorting necessary for non-ordered clones?

Reply:
Good question. It goes way back (5c9f91cbd15 in 2005!) without much explanation. Ordered wouldn't matter because only starts and stops are ordered, not probes.

Anonymous clones use only the first instance for probes, so I suppose sorting is the easiest way to find it (though it might be more efficient to create a :0 name and find that instance in the list). Unique clones will probe all children so it probably doesn't need to be sorted in that case.

Reply:
A quick test is promising but I'll leave this for later, when I can trace out the consequences better